frontend/src/app/utils/testResultCSV.ts

```diff
@@ -1,115 +1,149 @@
 
 import { displayFullName, facilityDisplayName } from "./index";
 
+export type QueriedTestResult = NonNullable<
+  NonNullable<
+    GetFacilityResultsForCsvWithCountQuery["testResultsPage"]
+  >["content"]
+>[number];
+
+export interface ResultCsvRow {
+  "Patient first name": string | null | undefined;
+  "Patient middle name": string | null | undefined;
+  "Patient last name": string | null | undefined;
+  "Patient full name": string | null | undefined;
+  "Patient date of birth": string | null | undefined;
+  "Test date": string | null | undefined;
+  Condition: string | null | undefined;
+  Result: string | null | undefined;
+  "Result reported date": string | null | undefined;
+  "Test correction status": string | null | undefined;
+  "Test correction reason": string | null | undefined;
+  "Device name": string | null | undefined;
+  "Device manufacturer": string | null | undefined;
+  "Device model": string | null | undefined;
+  "Device swab type": string | null | undefined;
+  "Has symptoms": string | null | undefined;
+  "Symptoms present": string | null | undefined;
+  "Symptom onset": string | null | undefined;
+  "Facility name": string | null | undefined;
+  Submitter: string | null | undefined;
+  "Patient role": string | null | undefined;
+  "Patient ID (Student ID, Employee ID, etc.)": string | null | undefined;
+  "Patient preferred language": string | null | undefined;
+  "Patient phone number": string | null | undefined;
+  "Patient email": string | null | undefined;
+  "Patient street address": string | null | undefined;
+  "Patient street address 2": string | null | undefined;
+  "Patient city": string | null | undefined;
+  "Patient state": string | null | undefined;
+  "Patient zip code": string | null | undefined;
+  "Patient county": string | null | undefined;
+  "Patient country": string | null | undefined;
+  "Patient gender": string | null | undefined;
+  "Patient race": string | null | undefined;
+  "Patient ethnicity": string | null | undefined;
+  "Patient tribal affiliation": string | null | undefined;
+  "Patient is a resident in a congregate setting": boolean | null | undefined;
+  "Patient is employed in healthcare": boolean | null | undefined;
+}
+
 export function parseDataForCSV(
-  data: TestResult[],
+  data: QueriedTestResult[],
   excludedDiseases: MULTIPLEX_DISEASES[] = []
-) {
-  return data.sort(byDateTested).map((r: any) => {
-    const symptomList = r.symptoms ? symptomsStringToArray(r.symptoms) : [];
+): ResultCsvRow[] {
+  let csvRows: ResultCsvRow[] = [];
+  data.sort(byDateTested).forEach((r: QueriedTestResult) => {
+    const symptomList = r?.symptoms ? symptomsStringToArray(r.symptoms) : [];
+
+    const swabTypes = r?.deviceType?.swabTypes ?? [];
 
-    const csvData1 = {
-      "Patient first name": r.patient.firstName,
-      "Patient middle name": r.patient.middleName,
-      "Patient last name": r.patient.lastName,
+    const csvData1: Partial<ResultCsvRow> = {
```

Comment:
nit: Are there more descriptive names for partial object typed variables that can tell us why we need to split them apart like this? Would be nice to have semantic names in the same way that you added `csvConditionData` that would help someone adding /modifying a column in the future know which / how to change these objects